pkg/controllers/resources/pods/translate/translator.go

```diff
@@ -959,3 +965,15 @@ func isInt64Different(i1, i2 *int64) (*int64, bool) {
 
 	return updated, false
 }
+
+func isPodSpecSchedulingGatesDiff(pGates, vGates []corev1.PodSchedulingGate) ([]corev1.PodSchedulingGate, bool) {
```

Comment:
Please just return bool here and then assign via `updatedPodSpec.SchedulingGates = vObj.Spec.SchedulingGates`